cmd/ctl/pkg/create/certificaterequest.go

```diff
@@ -0,0 +1,273 @@
+}
+
+// NewCmdCreateCertficate returns a cobra command for create CertificateRequest
+func NewCmdCreateCertficate(ioStreams genericclioptions.IOStreams, factory cmdutil.Factory) *cobra.Command {
+	o := NewOptions(ioStreams)
+	cmd := &cobra.Command{
+		Use:     "certificaterequest",
+		Aliases: alias,
+		Short:   "Create a CertificateRequest resource",
+		Long:    long,
+		Example: example,
+		Args: func(cmd *cobra.Command, args []string) error {
+			if len(args) > 1 {
+				return fmt.Errorf("only one argument can be passed in as the name of the certificate request")
+			}
+			return nil
+		},
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(o.Complete(factory))
+			cmdutil.CheckErr(o.Run(args))
+		},
+	}
+
+	cmdutil.AddFilenameOptionFlags(cmd, &o.FilenameOptions, "Path to a the manifest of Certificate resource.")
+
+	return cmd
+}
+
+// Complete takes the command arguments and factory and infers any remaining options.
+func (o *Options) Complete(f cmdutil.Factory) error {
+	var err error
+
+	err = o.FilenameOptions.RequireFilenameOrKustomize()
+	if err != nil {
+		return err
+	}
+
+	o.CmdNamespace, o.EnforceNamespace, err = f.ToRawKubeConfigLoader().Namespace()
+	if err != nil {
+		return err
+	}
+
+	o.RESTConfig, err = f.ToRESTConfig()
+	if err != nil {
+		return err
+	}
+
+	o.CMClient, err = cmclient.NewForConfig(o.RESTConfig)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Run executes create certificaterequest command
+func (o *Options) Run(args []string) error {
+	builder := new(resource.Builder)
+
+	// Read file as internal API version
+	r := builder.
+		WithScheme(scheme, schema.GroupVersion{Group: cmapiv1alpha2.SchemeGroupVersion.Group, Version: runtime.APIVersionInternal}).
+		LocalParam(true).ContinueOnError().
+		NamespaceParam(o.CmdNamespace).DefaultNamespace().
+		FilenameParam(o.EnforceNamespace, &o.FilenameOptions).Flatten().Do()
+
+	if err := r.Err(); err != nil {
+		return fmt.Errorf("error when getting Result from Builder: %s", err)
+	}
+
+	singleItemImplied := false
+	infos, err := r.IntoSingleItemImplied(&singleItemImplied).Infos()
+	if err != nil {
+		return fmt.Errorf("error when getting infos out of Result: %s", err)
+	}
+
+	// Ensure only one object per command
+	if len(infos) == 0 {
+		return fmt.Errorf("no object passed to create certificaterequest")
+	}
+	if len(infos) > 1 {
+		return fmt.Errorf("multiple objects passed to create certificaterequest")
+	}
+
+	info := infos[0]
+	// Convert to v1alpha2 because that version is needed for functions that follow
+	crtObj, err := scheme.ConvertToVersion(info.Object, cmapiv1alpha2.SchemeGroupVersion)
+	if err != nil {
+		return fmt.Errorf("failed to convert object into version v1alpha2: %v", err)
+	}
+
+	// Cast Object into Certificate
+	crt, ok := crtObj.(*cmapiv1alpha2.Certificate)
+	if !ok {
+		return fmt.Errorf("decoded object is not a v1alpha2 Certificate")
+	}
+
+	fmt.Println("Successfully decoded the object to a v1alpha2 Certificate")
```

Comment:
Would we want to print this for the end user?

Reply:
I thought it was a good point to signal nothing is going wrong. Do you think the user shouldn't see this?